### PR TITLE
EES-5424 Add deleted and new location group tables to location mappings page

### PR DESCRIPTION
**Pull request by ntsim**
This PR adds additional tables for deleted and new location groups. These types of changes occur if a set of location columns are changed in some way e.g. local authority columns are removed or added.

We don't consider location groups to be mappable to one another, so only add or delete operations are possible.

The UI looks like the following:

image

Note the 'Location groups not found in new data set' and 'New location groups' sections have been added.

Clicking the 'View group options' button for one of these shows:

image

## Other changes

- Fixed mapping error messages not linking to the correct parts of the page (locations and filters).
- Changed formatting of location code labels to use `code` elements. This was agreed with Marv although it hasn't been reflected in the prototype yet.
- Changed formatting of filter column ID labels to use `code` elements as well.

### CODE REVIEW

File: src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetMappableTable.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ export default function ApiDataSetMappableTable({
   return (
     <table
       className="dfe-table--vertical-align-middle dfe-table--row-highlights"
-      id={`mappable-filter-options-${kebabCase(groupKey)}`}
+      id={`mappable-table-${camelCase(groupKey)}`}
```
Comment (amyb-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetMappableTable.tsx:76]: Why the change from `kebabCase` to `camelCase` here? In the styleguide the preference is for kebab.

File: src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/ApiDataSetLocationGroupOptionsModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+import ApiDataSetLocationGroupOptionsModal from '@admin/pages/release/data/components/ApiDataSetLocationGroupOptionsModal';
+import { LocationOptionSource } from '@admin/services/apiDataSetVersionService';
+import render from '@common-test/render';
+import { screen, within } from '@testing-library/react';
+
+describe('ApiDataSetMappingModal', () => {
```
Comment (amyb-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/data/components/__tests__/ApiDataSetLocationGroupOptionsModal.test.tsx:6]: This should be 'ApiDataSetLocationGroupOptionsModal'

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx [hunk 3]
```diff
@@ -100,11 +115,27 @@ export default function ReleaseApiDataSetLocationsMappingPage() {
 
   const navItems: NavItem[] = useMemo(() => {
     return [
+      ...(Object.keys(deletedLocationGroups).length > 0
```
Comment (amyb-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx:118]: I didn't add the filter column equivalents for this and new location groups to the nav on the filters mapping page, but it probably makes sense to add them in to make it consistent with the locations page.

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx [hunk 4]
```diff
@@ -100,11 +115,27 @@ export default function ReleaseApiDataSetLocationsMappingPage() {
...
+        ? [
+            {
+              id: 'deleted-location-groups',
```
Comment (amyb-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx:121]: We've got a naming disparity between this and the filter columns, you've gone with 'deleted' whereas I went with 'mappable' on the basis that in the future you'll be able to map filter columns. Is the plan to be able to map location groups eventually? If so, it would be good to standardise the naming to avoid confusion.

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetFiltersMappingPage.tsx [hunk 5]
```diff
@@ -315,7 +315,7 @@ export default function ReleaseApiDataSetFiltersMappingPage() {
                   className="govuk-heading-l dfe-flex dfe-align-items--center"
                   id="mappable-filter-columns"
                 >
-                  Filter columns not found in the new data set{' '}
+                  Filter columns not found in new data set{' '}
```
Comment (amyb-hiveit) [inline src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetFiltersMappingPage.tsx:318]: This is entirely subjective, but I think this, and the other places you've changed it, read better with "the".

**Review by amyb-hiveit — CHANGES_REQUESTED** [review]
One small change requested and some queries.

File: src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx (hunk 4 above)
Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/data/ReleaseApiDataSetLocationsMappingPage.tsx:121]: I've gone with deleted as I don't think there's any plans to make these mappable. There's a number of cases where mapping isn't really possible e.g. you can't turn a school into an region.

Changes to location groups also aren't going to be common so it's generally more tolerable not to have mappings than it is for filter changes.

File: src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetMappableTable.tsx (hunk 1 above)
Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/data/components/ApiDataSetMappableTable.tsx:76]: I did this for consistency with location level keys (which are camelCased).

However, it's not really a big deal to go with kebab-casing so I'll switch back.

**Review by amyb-hiveit — APPROVED** [review]
